src/python/pants/cache/restful_artifact_cache.py

```diff
@@ -5,28 +5,157 @@
+      register('--requests-logging-level', choices=['WARNING', 'INFO', 'DEBUG'],
+               # Reduce the somewhat verbose logging of requests.
+               default='WARNING',
+               advanced=True,
+               help='The logging level to set the requests logger to.')
+      register('--max-connection-pools', type=int, default=cls._default_pool_size,
+               help='The max number of separate hosts to maintain urllib3 pools for. '
+                    'Corresponds to `pool_connections` in `requests.adapters.HTTPAdapter`.')
+      register('--max-connections-within-pool', type=int, default=cls._default_pool_size,
+               help='The max number of connections to retain within a single pool. '
+                    'Corresponds to `pool_maxsize` in `requests.adapters.HTTPAdapter`.')
+      register('--max-retries', type=int, default=cls._default_retries,
+               help='The max number of retries to perform for failed artifact cache requests. '
+                    'Corresponds to `max_retries` in in `requests.adapters.HTTPAdapter`.')
+      # TODO: raise an exception if these secondary retry limits exceed --max-retries (which will
+      # just have no effect)?
+      register('--max-retries-on-connection-errors', type=int, default=cls._default_retries,
+               help='The maximum number of retries to perform for requests which fail to connect.'
+                    '\n\n--max-retries takes precedence over this option, so if this number is '
+                    'greater than --max-retries, the additional retries are ignored.')
+      register('--max-retries-on-read-errors', type=int, default=cls._default_retries,
+               help='The maximum number of retries to perform for requests which fail after the '
+                    'request is sent to the server.'
+                    '\n\n--max-retries takes precedence over this option, so if this number is '
+                    'greater than --max-retries, the additional retries are ignored.')
+      register('--backoff-factor', type=float, default=0,
+               help='The backoff factor to apply between retry attempts. '
+                    'Set to 0 to disable backoff.')
+      register('--blocking-pool', type=bool,
+               help='Whether a connection pool should block instead of creating a new connection '
+                    'when the connection pool is already at its maximum size. '
+                    'Corresponds to `pool_block` in `requests.adapters.HTTPAdapter`.')
+      # TODO: Make a custom option type for timeout lengths which validates that the value is
+      # nonzero, and perhaps parameterized by time unit (seconds, ms, ns?)!
+      register('--slow-download-timeout-seconds', type=int, default=60,
+               help='The time to wait while downloading a cache artifact before printing a warning '
+                    'about a slow artifact download.')
+
+    @classmethod
+    def create(cls, logger):
```

Comment:
```suggestion
    def create(cls, logger) -> 'RequestsSession':
```

(Might be able to not use the quotes. Once we drop Py3.6, we can start using `from __future__ import annotations` and stop using the quotes. Hence why I'm eager to upgrade.)